pkg/autodiscovery/providers/prometheus_pods.go

```diff
@@ -0,0 +1,150 @@
+		check.init()
+	}
+
+	return nil
+}
+
+// String returns a string representation of the PrometheusPodsConfigProvider
+func (p *PrometheusPodsConfigProvider) String() string {
+	return names.Prometheus
+}
+
+// Collect retrieves templates from the kubelet's podlist, builds config objects and returns them
+func (p *PrometheusPodsConfigProvider) Collect() ([]integration.Config, error) {
+	var err error
+	if p.kubelet == nil {
+		p.kubelet, err = kubelet.GetKubeUtil()
+		if err != nil {
+			return []integration.Config{}, err
+		}
+	}
+
+	pods, err := p.kubelet.GetLocalPodList()
+	if err != nil {
+		return []integration.Config{}, err
+	}
+
+	return p.parsePodlist(pods), nil
+}
+
+// IsUpToDate always return false to poll new data from kubelet
+func (p *PrometheusPodsConfigProvider) IsUpToDate() (bool, error) {
+	return false, nil
+}
+
+// parsePodlist searches for pods that match the AD configuration
+func (p *PrometheusPodsConfigProvider) parsePodlist(podlist []*kubelet.Pod) []integration.Config {
+	configs := []integration.Config{}
+	for _, pod := range podlist {
+		for _, check := range p.checks {
+			configs = append(configs, check.configsForPod(pod)...)
+		}
+	}
+	return configs
+}
+
+// configsForPod returns the openmetrics configurations for a given pod if it matches the AD configuration
+func (pc *PrometheusCheck) configsForPod(pod *kubelet.Pod) []integration.Config {
+	configs := []integration.Config{}
+
+	for k, v := range pc.AD.KubeAnnotations.Excl {
+		if pod.Metadata.Annotations[k] == v {
+			log.Debugf("Pod '%s' matched the exclusion annotation '%s=%s' ignoring it", pod.Metadata.Name, k, v)
+			return configs
+		}
+	}
+
+	for k, v := range pc.AD.KubeAnnotations.Incl {
+		if pod.Metadata.Annotations[k] == v {
+			log.Debugf("Pod '%s' matched the annotation '%s=%s' to schedule an openmetrics check", pod.Metadata.Name, k, v)
+
+			instances := []integration.Data{}
+			for _, instance := range pc.Instances {
+				instanceValues := *instance
+				if instanceValues.URL == "" {
+					instanceValues.URL = buildURL(pod.Metadata.Annotations)
+				}
+				instanceJSON, err := json.Marshal(instanceValues)
+				if err != nil {
+					log.Warnf("Error processing prometheus configuration: %v", err)
+					continue
+				}
+				instances = append(instances, instanceJSON)
+			}
+
+			for _, container := range pod.Status.GetAllContainers() {
+				if !pc.AD.matchContainer(container.Name) {
+					log.Debugf("Container '%s' doesn't match the AD configuration 'kubernetes_container_names', ignoring it", container.Name)
+					continue
+				}
+				configs = append(configs, integration.Config{
+					Name:          openmetricsCheckName,
+					InitConfig:    integration.Data(openmetricsInitConfig),
+					Instances:     instances,
+					Provider:      names.Prometheus,
+					Source:        "kubelet:" + container.ID,
+					ADIdentifiers: []string{container.ID},
+				})
+			}
+
+			return configs
+		}
+	}
+
+	// TODO: Support AD matching based on label selectors
```

Comment:
Any difficulty to implement this? We already have necessary information from `pod`, it's just a matter of creating Incl/Excl parameters in the config no?

Reply:
yes indeed, will be implemented in another PR just keep a reasonable size for this PR